src/main/kotlin/com/openlattice/authorization/HazelcastAuthorizationService.kt

```diff
@@ -0,0 +1,526 @@
+package com.openlattice.authorization
+
+import com.codahale.metrics.annotation.Timed
+import com.google.common.base.Preconditions.checkNotNull
+import com.google.common.collect.HashMultimap
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.Maps
+import com.google.common.collect.SetMultimap
+import com.google.common.eventbus.EventBus
+import com.hazelcast.aggregation.Aggregators
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.core.IMap
+import com.hazelcast.query.Predicate
+import com.hazelcast.query.Predicates
+import com.openlattice.assembler.events.MaterializePermissionChangeEvent
+import com.openlattice.authorization.aggregators.AuthorizationSetAggregator
+import com.openlattice.authorization.aggregators.PrincipalAggregator
+import com.openlattice.authorization.mapstores.PermissionMapstore
+import com.openlattice.authorization.mapstores.PermissionMapstore.ACL_KEY_INDEX
+import com.openlattice.authorization.processors.AuthorizationEntryProcessor
+import com.openlattice.authorization.processors.PermissionMerger
+import com.openlattice.authorization.processors.PermissionRemover
+import com.openlattice.authorization.processors.SecurableObjectTypeUpdater
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.authorization.util.toAceKeys
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.organizations.PrincipalSet
+import org.slf4j.LoggerFactory
+import java.time.OffsetDateTime
+import java.util.*
+import java.util.function.Function
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+
+class HazelcastAuthorizationService(
+        hazelcastInstance: HazelcastInstance,
+        eventBus: EventBus
+) : AuthorizationManager {
+
+    private var securableObjectTypes: IMap<AclKey, SecurableObjectType> = HazelcastMap.SECURABLE_OBJECT_TYPES.getMap(hazelcastInstance)
+    private var aces: IMap<AceKey, AceValue> = HazelcastMap.PERMISSIONS.getMap(hazelcastInstance)
+    private var eventBus: EventBus = checkNotNull(eventBus)
```

Comment:
I don't think you need the checkNotNull anymore b/c eventbus isn't nullable. Probably also want to make eventBus constructor arg a val and make the IMaps vals as well

Reply:
➕